drivers/w1/w1_shell.c

```diff
@@ -0,0 +1,413 @@
+	int ret;
+
+	dev = device_get_binding(argv[pos]);
+	if (!dev) {
+		shell_error(sh, W1DEV_X_NOT_FOUND, argv[1]);
+		return -EINVAL;
+	}
+	pos++;
+
+	pos = read_io_options(sh, pos, argv, &reset);
+	if (pos < 0) {
+		return -EINVAL;
+	}
+	if (argc <= pos) {
+		shell_error(sh, "Missing data to be written.");
+		return -EINVAL;
+	}
+	if ((argc - pos) > BUF_SIZE) {
+		shell_error(sh, "Too much data to be written.");
+		return -EINVAL;
+	}
+
+	(void)w1_lock_bus(dev);
+	i = 0;
+	do {
+		msg_buf[i] = (uint8_t)strtoul(argv[i + pos], NULL, 16);
+		i++;
+	} while ((i + pos) < argc);
+
+	if (reset) {
+		ret = w1_reset_bus(dev);
+		if (ret <= 0) {
+			shell_error(sh, "Failed to reset bus [%d]", ret);
+			goto out;
+		}
+	}
+
+	ret = w1_write_block(dev, msg_buf, i);
+	if (ret < 0) {
+		shell_error(sh, "Failed to write block [%d]", ret);
+	}
+out:
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+/* 1-Wire config <device> <type> <value> */
+static int cmd_w1_configure(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	char *type_endptr;
+	char *type_name = argv[2];
+	int ret;
+	uint32_t type = strtoul(type_name, &type_endptr, 0);
+	uint32_t value = strtoul(argv[3], NULL, 0);
+
+	dev = device_get_binding(argv[1]);
+	if (!dev) {
+		shell_error(sh, W1DEV_X_NOT_FOUND, argv[1]);
+		return -EINVAL;
+	}
+
+	/* if type is not given as number, search it via the name */
+	if (*type_endptr != '\0') {
+		for (type = 0; type < ARRAY_SIZE(w1_settings_name); type++) {
+			if (strcmp(type_name, w1_settings_name[type]) == 0) {
+				break;
+			}
+		}
+
+		if (type == ARRAY_SIZE(w1_settings_name)) {
+			shell_error(sh, "Unknown config name (%s)", type_name);
+			return -ENOTSUP;
+		}
+	}
+
+	if (type > W1_SETINGS_TYPE_COUNT) {
+		shell_error(sh, "invalid type %u", type);
+		return -EINVAL;
+	}
+
+	(void)w1_lock_bus(dev);
+	ret = w1_configure(dev, type, value);
+	if (ret < 0) {
+		shell_error(sh, "Failed to configure [%d]", ret);
+		goto out;
+	}
+
+	shell_info(sh, "Applied config: %s = %u (0x%08x)", w1_settings_name[type], value, value);
+
+out:
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+static void search_callback(struct w1_rom rom, void *user_data)
+{
+	const struct shell *sh = (const struct shell *)user_data;
+
+	shell_print(sh, "ROM found: %016llx", w1_rom_to_uint64(&rom));
+}
+
+/* 1-Wire search <device> */
+static int cmd_w1_search(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int ret;
+
+	dev = device_get_binding(argv[1]);
+	if (!dev) {
+		shell_error(sh, W1DEV_X_NOT_FOUND, argv[1]);
+		return -EINVAL;
+	}
+
+	(void)w1_lock_bus(dev);
+	ret = w1_search_rom(dev, search_callback, (void *)sh);
+	if (ret < 0) {
+		shell_error(sh, "Failed to initiate search [%d]", ret);
+	} else {
+		shell_print(sh, "Found %d device(s)", ret);
+	}
+
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_w1,
+	SHELL_CMD_ARG(reset, NULL,
+		     "Reset 1-Wire bus.\n"
+		     "<device_name>",
```

Comment:
```suggestion
		     "<device_label>",
```
This makes it more obvious that this is the `label` field and not the node name.

Reply:
changed <device_name> to <device> in the help text.